llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -465,6 +465,16 @@ static bool isProjectedMap(AffineMap map, int64_t projectedDim) {
   return true;
 }
 
+/// Helper to turn a potentially negative index to positive within the range [0,
+/// ub] and indicate whether the transformed index is in bounds.
```

Comment:
Ultra-nit: I think you'd want to specify that `ub` is not included in the range as either `[0, ub)` or somehow stating that explicitly.

Reply:
Oh no, I flipped the logic here once and forgot to update the comment. I opened #13546 to address the comments here. In the future I'll make sure to give more time for review, apologies this time though :(